x-pack/plugins/apm/public/components/app/ErrorGroupOverview/index.tsx

```diff
@@ -55,9 +56,10 @@ const ErrorGroupOverview: React.SFC<ErrorGroupOverviewProps> = ({
         end,
         sortField,
         sortDirection,
-        kuery
+        kuery,
+        environment
       }),
-    [serviceName, start, end, sortField, sortDirection, kuery]
+    [serviceName, start, end, sortField, sortDirection, kuery, environment]
```

Comment:
I don't think it'll scale passing every single filter (global and local) individually around in the application. Instead we should have a `uiFilters` object which contains both global (kuery, environment) and local (transaction type, transaction result, status code etc) filters. 

Filters should still be added to the url individually (flat, no Rison) so they are linkable - perhaps prefixed with `gfilter.environment` and `lfilter.transactiontype` to denote whether it is a global or local filter.

In `urlParams` all the global and local filters should be consolidated into a single object (`uiFilters`) which is basically just:

```js
{
 global: {
   kuery: 'transaction.duration.us > 100',
   environment: 'prod'
 },
 local: {
  transactionType: 'request'
 }
}
```

This can be consumed from urlParams like:
```js
const { uiFilters } = useUrlParams();
```

**Backend**
To apply `uiFilters` to an ES query, we need to convert it to a list of boolean ES filters. 
```js
const uiFiltersES = 
```

Where `uiFiltersES` is:
```js
[
  // kuery
  { range: { 'transaction.duration.us': { gt: 100 } } },

  // environment
  { term: { 'service.environment': 'production' } },

  // transaction type
  { term: { 'transaction.type': 'request' } },
];
```

Today we simply apply the kuery filter to ES queries like:
https://github.com/elastic/kibana/blob/87b0fd1158c3ecba28444d61d0ef6fd307b82819/x-pack/plugins/apm/server/lib/errors/get_error_group.ts#L41-L43

We should replace that with:
```
  if (uiFiltersES) {
    filter.push(...uiFiltersES);
  }
```